source/initial_temperature/lithosphere_mask.cc

```diff
@@ -0,0 +1,195 @@
+          else
+            temperature = std::numeric_limits<double>::quiet_NaN();
+        }
+
+      else if (LAB_depth_source == Value)
+        {
+          if (depth <= max_depth*1000)//in km
+            temperature = lithosphere_temperature;
+          else
+            temperature = std::numeric_limits<double>::quiet_NaN();
+        }
+
+      else
+        {
+          Assert( false, ExcMessage("Invalid method for depth specification method") );
+          return 0.0;
+        }
+
+      return temperature;
+    }
+
+    template <int dim>
+    void
+    LithosphereMask<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection ("Lithosphere Mask");
+        {
+          prm.declare_entry ("Lithosphere temperature", "1600",
+                             Patterns::Double (0),
+                             "The initial temperature within lithosphere, applied above"
+                             "the maximum lithosphere depth.");
+          prm.declare_entry ("Depth specification method", "Value",
+                             Patterns::Selection("File|Value"),
+                             "Method that is used to specify the depth of the lithosphere-asthenosphere boundary.");
+          prm.declare_entry ("Maximum lithosphere depth", "200.0",
+                             Patterns::Double (0),
+                             "The maximum depth of the lithosphere. The model will be "
+                             "NaNs below this depth.");
+          prm.declare_entry ("Data directory", "$ASPECT_SOURCE_DIR/data/initial-temperature/lithosphere_mask/",
+                             Patterns::DirectoryName (),
+                             "The path to the LAB depth data file");
+          prm.declare_entry ("LAB depth filename",
+                             "LAB_CAM2016.txt",
+                             Patterns::FileName (),
+                             "File from which the lithosphere-asthenosphere boundary depth data is read.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+
+
+    LithosphereMask<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      AssertThrow (dim == 3,
+                   ExcMessage ("The 'Lithosphere mask' model for the initial "
+                               "temperature is only available for 3d computations."));
+
+      prm.enter_subsection("Initial temperature model");
+      {
+        prm.enter_subsection ("Lithosphere Mask");
+        {
+          lithosphere_temperature   = prm.get_double ("Lithosphere temperature");
+
+          if ( prm.get("Depth specification method") == "File" )
+            {
+              LAB_depth_source = File;
+              data_directory = Utilities::expand_ASPECT_SOURCE_DIR (prm.get("Data directory"));
+              LAB_file_name   = prm.get("LAB depth filename");
```

Comment:
please either align the `=` signs, or put them with a single space after the name of the variable (your choice). Just nitpicking, but the current spacing looks weird.